Automattic/wp-calypso
# Signup:  add 'site-information' step to onboarding flow

This PR adds the `site-information` step component for the new `/onboarding/` Signup flow.

This step appears along with the site type AB test which was introduced in #28074. With the merge of the `site-type` and `site-topic` steps, this PR removes the `about` step from the onboarding flow, since all fields in that step are now covered in different steps.

## Design

Site type: Business | Site type: Blog
------------ | -------------
<img width="862" alt="business site information" src="https://user-images.githubusercontent.com/1269602/48356162-a6696d00-e6bb-11e8-8858-016d551b36b5.png"> | <img width="862" alt="blog site information" src="https://user-images.githubusercontent.com/1269602/48356231-cc8f0d00-e6bb-11e8-9c7d-097bb957df18.png">

At this point, the order of steps in the `/onboarding/` flow should be:

1. User
2. Site Type
3. Site Topic
4. Site Information
5. Domain
6. Select Plan

## Dependencies
~❗️ #28163 should be merged into master first for the redux state files~
~❗️ #28074 should be merged into master first as we're relying on `siteType`~
Both #28163 and #28074 are now merged into master

## TODO
- [x] Add styles
- [x] Rebase after #28163 and #28074 are merged
- [x] Add step to flow

# Testing

### Activating the AB test

Navigate to `/start/user` and run the command below in your browser's console. You might have to reload the page after running both commands:

```
localStorage.setItem( 'debug', 'calypso:abtests' );
localStorage.setItem( 'ABTests', '{"signupSegmentationStep_20181029":"include", "improvedOnboarding_20181023":"onboarding"}' );
```
### De-activating the AB test
```
localStorage.setItem( 'ABTests', '{"signupSegmentationStep_20181029":"exclude", "improvedOnboarding_20181023":"onboarding"}' );
```

## Testing the flow

### Scenario 1
1. Visit /start and commence the new site/sign up process
2. At the _Site Type_ step, select Educator
3. In the _About_ step, verify that the Site Name field is hidden
4. In the Site Information, verify that the field name says _Site Name_
5. After completing the signup and logging in to your console, verify that the checklist item for Site Title is marked completed:
<img width="760" alt="screenshot 2018-11-12 at 8 59 47 pm" src="https://user-images.githubusercontent.com/1269602/48357160-eaf60800-e6bd-11e8-84f1-4650518aa7ba.png">
6. Verify that the site name you entered in the *Site Information* step appears correctly in General settings: `/settings/general/{siteSlug}`

### Scenario 2
For the most parts, the same as above. A few differences:
1. Select *Site Type* as Business
2. Verify that the Site Information step asks for Business Name, address, email and phone number
3. Verify that email, address and phone number fields you entered in *Site Information* step appear in your newly created site's Contact page in Calypso preview.

#### ~~TODO / Known Issue~~
~~1. The Address/phone number/email fields entered in the Site Information step are not yet mapped to headstart, so you will not see them getting populated in the Contact section of your site in Calypso Preview.~~

### Scenario 3 (Testing headstart)

1. Apply code-D20806 to your sandbox
2. Visit /start and commence the signup process with the signupSegmentationStep AB test included.
3. Select `Business` in **Site Type**, and then in the **Site Information** step input the business name, address and phone details.
4. Complete the signup process and launch into Calypso. Go to the Calypso preview for your site:
`/view/{siteSlug}` and verify that the business name and contact details entered in the Site Informations step are shown here. It should look something like this:

<img width="416" alt="screenshot 2018-11-15 at 9 15 53 pm" src="https://user-images.githubusercontent.com/1269602/48603940-2171a280-e99e-11e8-815b-80e724d5fe6d.png">
